package/MDAnalysis/analysis/base.py

```diff
@@ -447,8 +494,177 @@ def run(self, start=None, stop=None, step=None, frames=None,
+            bsizes += np.arange(n_bslices, dtype=np.int64) < n_frames % n_bslices
+            # This can give a last index that is larger than the real last index;
+            # this is not a problem for slicing but it's not pretty.
+            # Example: original [0:20:3] -> n_frames=7, start=0, step=3:
+            #          last frame 21 instead of 20
+            bsizes *= step
+            idx = np.cumsum(np.concatenate(([start], bsizes)))
+            slices = [(bstart, bstop, step, None) 
+                       for bstart, bstop in zip(idx[:-1], idx[1:])]
+
+            # fix very last stop index: make sure it's within trajectory range or None
+            # (no really critical because the slices will work regardless, but neater)
+            last = slices[-1]
+            last_stop = min(last[1], stop) if stop is not None else stop
+            slices[-1] = (last[0], last_stop, last[2], None)
+        
+        self._bslices = slices
+        return self._bslices
+
+
+    def _setup_scheduler(self, scheduler, n_workers):
+        """
+        Configure parameters necessary for running a distributed workload.
+
+        Parameters
+        ----------
+        scheduler : dask.distributed.Client
+            dask scheduler object
+        n_workers : int, optional
+            number of workers (local or remote processes).
+        """
+        if scheduler == 'localdask':
+            n_workers = 1
+        else:
+            from dask.distributed import Client
+            if isinstance(scheduler, Client):
+                # TODO: add assertions that make sure that 
+                # you don't set 
+                # both `scheduler` and `n_workers/threads_per_worker`
+                n_workers = len(scheduler.ncores())
+            else:
+                kwargs = {'n_workers':n_workers}
+            self._scheduler_kwargs = kwargs
+        self._n_bslices = self._n_workers = n_workers
+
+    def run(self, start=None, stop=None, step=None, frames=None,
+            verbose=None, n_workers=1, 
+            *, scheduler=None,
+            progressbar_kwargs={},
+            ):
+        """Perform the calculation
+
+        Parameters
+        ----------
+        start : int, optional
+            start frame of analysis
+        stop : int, optional
+            stop frame of analysis
+        step : int, optional
+            number of frames to skip between each analysed frame
+        frames : array_like, optional
+            array of integers or booleans to slice trajectory; `frames` can
+            only be used *instead* of `start`, `stop`, and `step`. Setting
+            *both* `frames` and at least one of `start`, `stop`, `step` to a
+            non-default value will raise a :exc:`ValueError`.
+
+            .. versionadded:: 2.2.0
+
+        verbose : bool, optional
+            Turn on verbosity
+        
+        scheduler : str, optional
+            Enables running with different schedulers.
```

Comment:
not clear what the `scheduler` kwarg really does — if you don't know dask then it makes zero sense. Should be made clear that it's related to parallelization and `n_workers`.

`n_workers` should be explained, too.

Imagine reading the docs as someone who is not you... will that person be able to discover new functionality or learn enough to make use of your new features?

Reply:
I believe the docs are now clearer than they used to be, thanks.